cardshifter-server/src/test/java/com/cardshifter/server/main/ServerConnectionTest.java

```diff
@@ -252,5 +252,16 @@ public void testUsersQueryNotLoggedIn() throws IOException, InterruptedException
 		// There shouldn't be a UserStatusMessage for client2
 		assertNoMessage(client1);
 	}
+
+	@Test(timeout = 5000)
+	public void testNoOfflineMessageIfNotLoggedIn() throws IOException, InterruptedException {
+		System.out.println("testNoOfflineMessageIfNotLoggedIn");
+		TestClient client2 = createTestClient();
+		client2.disconnect();
+
+		// Avoid race condition between UserStatusMessage and ServerStatusMessage sent by assertNoMessage
+		Thread.sleep(1000);
```

Comment:
I'm not sure using `Thread.sleep()` to fudge this feature working is the best way to approach this. Maybe it would be better to solve the underlying error before the entire solution revolves around `Thread.sleep`.